cpp/src/io/orc/reader_impl.cu

```diff
@@ -425,7 +442,15 @@ table_with_metadata reader::impl::read(size_type skip_rows,
   // Get a list of column data types
   std::vector<data_type> column_types;
   for (const auto &col : _selected_columns) {
-    auto col_type = to_type_id(_metadata->ff.types[col], _use_np_dtypes, _timestamp_type.id());
+    // If the column type is orc::DECIMAL see if the user
+    // desires it to be converted to float64 or not
+    bool decimal_as_float64 = false;
+    if (_metadata->ff.types[col].kind == orc::DECIMAL &&
+        convert_decimal_column_to_float(_decimal_cols_as_float, _metadata, col))
+      decimal_as_float64 = true;
```

Comment:
We can move the type check to `convert_decimal_column_to_float` and simplify this code a bit
```suggestion
    auto const decimal_as_float64 = convert_decimal_column_to_float(_decimal_cols_as_float, _metadata, col));
```